### PR TITLE
[ThinLTO]Sort imported GUIDs before cache key update

### PR DESCRIPTION
**Pull request by mingmingl-llvm**
@MaskRay points out `std::` doesn't randomize the iteration order of `unordered_{set,map}`, and the iteration order for single build is deterministic.

Specifically, add 'sort' here since it's helpful when container type changes (for example, https://github.com/llvm/llvm-project/pull/88024 wants to change container type from `unordered_set` to `DenseMap)

### CODE REVIEW

File: llvm/lib/LTO/LTO.cpp [hunk 1]
```diff
@@ -199,13 +199,21 @@ void llvm::computeLTOCacheKey(
...
 
     AddUint64(Entry.getFunctions().size());
+
+    ImportedGUIDs.clear();
     for (auto &Fn : Entry.getFunctions())
-      AddUint64(Fn);
+      ImportedGUIDs.push_back(Fn);
+
+    llvm::sort(ImportedGUIDs);
+
```
Comment (MaskRay) [inline llvm/lib/LTO/LTO.cpp:214]: There are quite a few blank lines but I think we can be more dense.

Group coherent code block of multiple lines, probably not just one/two lines.

**Review by MaskRay — APPROVED** [review]
Thanks! https://abseil.io/docs/cpp/guides/container#iteration-order-instability and DenseMap/StringMap "randomize" the iteration order to catch mistakes, but libstdc++/libc++ unordered_{set,map} haven't implemented this technique. (Many simple concepts are extremely difficult to implement portably in libc++...)

File: llvm/lib/LTO/LTO.cpp (hunk 1 above)
Comment (minglotus-6) [reply llvm/lib/LTO/LTO.cpp]: thanks for the pointers!

**Comment by minglotus-6** [comment]
Test failure is irrelevant.

```
cat github-pull-requests_build_65187_linux-linux-x64.log | grep -B 5 -A 10 "Failed Tests"
[0.20s,0.30s) :: [***********                             ] :: [120/418]
[0.10s,0.20s) :: [******************                      ] :: [191/418]
[0.00s,0.10s) :: [*******                                 ] :: [ 74/418]
--------------------------------------------------------------------------
********************
Failed Tests (1):
  BOLT :: RISCV/unnamed-sym-no-entry.c


Testing Time: 2.47s

Total Discovered Tests: 431
  Skipped          :   7 (1.62%)
  Unsupported      :  13 (3.02%)
  Passed           : 409 (94.90%)
  Expectedly Failed:   1 (0.23%)
```

**Pull request merged**